src/document.rs

```diff
@@ -51,24 +51,22 @@ pub(crate) fn extract_value(any: &PyAny) -> PyResult<Value> {
     if let Ok(b) = any.extract::<Vec<u8>>() {
         return Ok(Value::Bytes(b));
     }
-    if let Ok(dict) = any.downcast::<PyDict>() {
-        if let Ok(json) = pythonize::depythonize(dict) {
-            return Ok(Value::Object(json));
-        }
+    if let Ok(json) = pythonize::depythonize_bound(any.clone()) {
```

Comment:
The first of several `.clone()` calls in this PR.

If I understand [the documentation](https://pyo3.rs/v0.21.2/migration#migrating-from-the-gil-refs-api-to-boundt) correctly:

> Code will need to add the occasional & to borrow the new smart pointer as &Bound<T> to pass these types around (or use .clone() at the very small cost of increasing the Python reference count)

So based on this my understanding is that `Bound<_, T>.clone()` is cheap (inc refcount) and so this is what I did to get the code to compile. However, I am super unfamiliar with the new PyO3 API so it's entirely possible I've missed something.

I will mark all the new calls to `.clone()` in this PR.

Reply:
> So based on this my understanding is that Bound<_, T>.clone() is cheap (inc refcount) and so this is what I did to get the code to compile. However, I am super unfamiliar with the new PyO3 API so it's entirely possible I've missed something.

Yes, `clone` is indeed cheap as it only increments the reference count. In really performance critical code, I can be helpful to avoid even that cost especially if it ends up ineffective, e.g. temporary clones which increment and then decrement the reference count in a loop. But usually, it's fine.

In this case, the old `depythonize` took a GIL-bound reference `&PyAny`, so it borrowed the reference in Python terms. But this was only possible/safe because we kept a global data structure around which would keep track of any necessary reference count changes and would apply them before releasing the GIL. You can still get the same performance by passing around `&Bound<'_, PyAny>`, but now you can avoid the cost of the global data structure entirely and get full control over any reference count changes.

Reply:
Yes, you must clone here.